### ISSUE TITLE
Incompatibility with Ruby Versions older than 3.0

### ISSUE THREAD

**Issue by caitlincoffey** [markdownlint/markdownlint#521]
## Description
Due to a recent change with one of the markdownlint dependencies, mixlib-shellout, every version that I use of markdownlint must use Ruby version 3.0 or greater. An error is received when attempting to run it with a Ruby version older than 3.0.

### Environment
Linux CI and macOS.
**MDL Version**
0.10.0, 0.11.0, 0.12.0, 0.13.0

### Expected Behavior
No errors or abnormal logs should be produced.

### Actual Behavior
When linking this in a `.pre-commit-config.yaml` file, I received this error:
```
X: An unexpected error has occured: CalledProcessErrror: command: ('user/bin/bash', '/opt/.rbenv/shims/gem', 'install', '--no-document', '--no-format-executable', '--no-user-install', '--install-dir', '/home/git/.cahce/pre-commit/repovt4t1o1y/rbenv-default/gems', '--bindir', '/home/git/.cache/pre-commit/repovt4t1o1y/rbenv-default/gems/bin', 'mdl-0.11.0.gem')
X: return code: 1
X: stdout: Successfully installed concurrent-ruby-1.3.5
X: Successfully installed chef-utils-18.7.3
X: stderr: ERROR: Error installing mdl-0.11.0.gem: mixlib-shellout requires Ruby version >=3.0. The current ruby version is 2.7.8.225.
```

## Replication Case
I included markdownlint in my pre-commit file using the below code:

```
- repo: https://github.com/markdownlint/markdownlint
   rev: v0.11.0
   hooks:
      - id: markdownlint
         name: "Check Markdown files [markdownlint]"
         args: [-s, .markdownlint.rb]
````

My markdown.rb file contains the following listed content:
```
all
exclude_rule="MD013"
exclude_rule="MD033"
exclude_rule="MD034"
exclude_rule="MD041"
```

This command was ran using `pre-commit run --all-files`, see [https://pre-commit.com](https://pre-commit.com).

The version of Ruby that I am using is 2.7.8.225.

**Comment by v8p1197** [comment]
Hi, I have a similar error with Ruby 2.7.0.0 and markdownlint v0.12.0.
